R/sample.R

```diff
@@ -34,47 +44,107 @@ mcstate_sample <- function(model, sampler, n_steps, initial = NULL) {
     cli::cli_abort("Expected 'sampler' to be an 'mcstate_sampler'",
                    arg = "sampler")
   }
+  if (is.null(runner)) {
+    runner <- mcstate_runner_serial()
+  }
+  if (!inherits(runner, "mcstate_runner")) {
+    cli::cli_abort("Expected 'runner' to be an 'mcstate_runner'",
+                   arg = "runner")
+  }
+
+  rng <- initial_rng(n_chains)
+  pars <- initial_parameters(initial, model, rng, environment())
+  res <- runner$run(pars, model, sampler, n_steps, rng)
+  combine_chains(res)
+}
 
-  ## We might change this later.
-  rng <- mcstate_rng$new()
-  r_rng_state <- get_r_rng_state()
 
+mcstate_sampler <- function(name, initialise, step, finalise) {
+  ret <- list(name = name,
+              initialise = initialise,
+              step = step,
+              finalise = finalise)
+  class(ret) <- "mcstate_sampler"
+  ret
+}
+
+
+initial_parameters <- function(initial, model, rng, call = NULL) {
```

Comment:
It looks like this method will check initial parameters we generate from `model$direct_sample` as well as user-provided ones, which perhaps we don't need to do if we trust ourselves (but maybe we don't). Should we check for silly values like NULL (?) as well as for the right number of values?

Reply:
> we generate from model$direct_sample as well as user-provided ones

parameters from `direct_sample` are also really user defined as they've written that function and there's no guarantee that these are the right length.  `NULL` is covered by a check for length, though we could add a numeric check in too

Reply:
Ah yeah, that's true!